src/Microsoft.AspNet.Mvc.Localization/ViewLocalizer.cs

```diff
@@ -93,12 +96,30 @@ public void Contextualize(ViewContext viewContext)
                 throw new ArgumentNullException(nameof(viewContext));
             }
 
-            var baseName = viewContext.View.Path.Replace('/', '.').Replace('\\', '.');
-            if (baseName.StartsWith(".", StringComparison.OrdinalIgnoreCase))
+            // Given a view path "/Views/Home/Index.cshtml" we want a baseName like "MyApplication.Views.Home.Index"
+
+            var path = viewContext.ExecutingFilePath;
+
+            if (string.IsNullOrEmpty(path))
+            {
+                path = viewContext.View.Path;
+            }
+
+            // Trim the file extension from the end of the path
+            if (!string.IsNullOrEmpty(path) && Path.HasExtension(path))
             {
-                baseName = baseName.Substring(1);
+                var extension = Path.GetExtension(path);
+                path = path.Substring(0, path.Length - extension.Length);
             }
 
+            Debug.Assert(!string.IsNullOrEmpty(path), "Couldn't determine a path for the view");
```

Comment:
Just a reminder that this is here, fine to leave it but don't know if it's intentional.